File: magellan-library/src/main/java/com/wealthfront/magellan/Preconditions.java
package com.wealthfront.magellan;

public final class Preconditions {

  private Preconditions() {
    throw new AssertionError();
  }

  public static <T> T checkNotNull(T reference) {
    if (reference == null) {
      throw new NullPointerException();
    }
    return reference;
  }

  public static <T> T checkNotNull(T reference, String message) {
    if (reference == null) {
      throw new NullPointerException(message);
    }
    return reference;
  }

  public static void checkArgument(boolean expression) {
    if (!expression) {
      throw new IllegalArgumentException();
    }
  }

  public static void checkArgument(boolean expression, String message) {
    if (!expression) {
      throw new IllegalArgumentException(message);
    }
  }

  public static void checkState(boolean expression) {
    if (!expression) {
      throw new IllegalStateException();
    }
  }

  public static void checkState(boolean expression, String message) {
    if (!expression) {
      throw new IllegalStateException(message);
    }
  }

}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/Preconditions.kt
package com.wealthfront.magellan.kotlin

internal class Preconditions private constructor() {

  init {
    throw AssertionError()
  }

  companion object {

    fun <T> checkNotNull(reference: T?): T {
      if (reference == null) {
        throw NullPointerException()
      }
      return reference
    }

    fun <T> checkNotNull(reference: T?, message: String): T {
      if (reference == null) {
        throw NullPointerException(message)
      }
      return reference
    }

    fun checkArgument(expression: Boolean) {
      if (!expression) {
        throw IllegalArgumentException()
      }
    }

    fun checkArgument(expression: Boolean, message: String) {
      if (!expression) {
        throw IllegalArgumentException(message)
      }
    }

    fun checkState(expression: Boolean) {
      if (!expression) {
        throw IllegalStateException()
      }
    }

    fun checkState(expression: Boolean, message: String) {
      if (!expression) {
        throw IllegalStateException(message)
      }
    }
  }

}
